TrailheadApex11/ExercisesPage/src/classes/ExercisesController.cls

```diff
@@ -0,0 +1,85 @@
+
+public with sharing class ExercisesController {
+    public String inputExerciseString { get; set; }
+    public String outputExerciseString { get; set; }
+    public String infoExerciseString { get; set; }
+    public String className { get; set; }
+
+
+    public List<SelectOption> getItems() {
+        List<SelectOption> options = new List<SelectOption>();
+        options.add(new SelectOption('', '---Select---'));
+        options.add(new SelectOption('SayingHello', 'Saying Hello'));
+        options.add(new SelectOption('CountingNumbers', 'Counting the Number of Characters'));
+        options.add(new SelectOption('PrintingQuotes', 'Printing Quotes'));
+        options.add(new SelectOption('MadLib', 'Mad Lib'));
+        options.add(new SelectOption('SimpleMath', 'Simple Math'));
+        options.add(new SelectOption('RetirementCalculator', 'Retirement Calculator'));
+
+        return options;
+    }
+
+    public void solve() {
+        if (className == null) {
+            outputExerciseString = 'You have to choose the exercise! ';
+            return;
+        }
+        List<String> splittedInputString = inputExerciseString.split('\n');
+
+        ExercisesLoader exercise;
+
+        if (className.equals('SayingHello')) {
+            exercise = new Ex1SayingHello();
+        } else if (className.equals('CountingNumbers')) {
+            exercise = new Ex2CountingNumbers();
+        } else if (className.equals('PrintingQuotes')) {
+            exercise = new Ex3PrintingQuotes();
+        } else if (className.equals('MadLib')) {
+            exercise = new Ex4MadLib();
+        } else if (className.equals('SimpleMath')) {
+            exercise = new Ex5SimpleMath();
+        } else if (className.equals('RetirementCalculator')) {
+            exercise = new Ex6RetirementCalculator();
+        }
+
+        outputExerciseString = exercise.solveTheExercise(splittedInputString);
```

Comment:
Ja bym tu trochę zmenił design. 
2 propozycje:
a) niech solveTheExercise zwraca voida i dodaj metodę isSucces i metodę do pobierania treści wyniku
b) niech solveTheExercise zwraca boola od razu, będziesz mógł sprawdzić result w momencie wywołania i wtedy też użyć metody do pobrania wyniku.

Nie wiem czy tak jest jakoś szczególnie lepiej - ja bym tak zrobił po prostu ;0

Btw, raczej nie używamy 'the' w nazwach, jebać gramatykę, zrozumieć rozumiesz a tylko je to wydłuża. Nie jest to technicznie błąd, ale dla kontekstu sprawdziłem u mnie nazwy klas - na wchuj klas ani jedna nie ma w nazwie 'the'.